tools/mksurfdata_esmf/gen_mksurfdata_namelist.py

```diff
@@ -433,14 +467,24 @@ def main ():
                     if child2.tag == 'ny':
                         rawdata_files["mksrf_fgrid_mesh_ny"] = child2.text
 
-    if force_model_mesh_file == 'none' and len(model_mesh) == 0:
+    if not model_mesh and force_model_mesh_file is None:
         valid_grids = []
         for child1 in root:  # this is domain tag
             for _, value in child1.attrib.items():
                 valid_grids.append(value)
-        error_msg = f'ERROR: invalid input res {res};' \
-                    f'valid grid values are {valid_grids}'
-        sys.exit(error_msg)
+        if res in valid_grids:
+            error_msg = 'ERROR: You have requested a valid grid for which ' \
+        '../../ccs_config/component_grids_nuopc.xml does not include a mesh ' \
+        'file. For a regular regional or 1x1 grid, you may generate the ' \
+        'fsurdat file using the subset_data tool instead. Alternatively ' \
+        'and definitely for curvilinear grids, you may generate ' \
+        'a mesh file using the workflow currently (2022/7/6) described in ' \
+        'https://github.com/ESCOMP/CTSM/issues/1773#issuecomment-1163432584'
```

Comment:
@ekluzek comment:
add TODO that this workflow should be included in the User's Guide.
add similar comment in the issue.